### ISSUE TITLE
[#feature] Implement ES7 async wait

### ISSUE THREAD

**Issue by Globegitter** [balderdashy/sails#1926]
I have read some news from the Ember Data core team today and they were mentioning the possibility of including the new async wait functionality that is planned to come with ES7. Spending half an hour or so looking into the matter, this new feature looks very exciting and that it would make working with asynchronous functions/data much nicer, simpler and cleaner.
I know v0.10 is just about to ship so this would be a feature for a later state. Anyway, here is an article getting into the basics and showing the differences to ES6 promises:

http://jakearchibald.com/2014/es7-async-functions/

Here is the relevant ES7 proposal: https://github.com/lukehoban/ecmascript-asyncawait
And it seems the feature is already available using the traceur-compiler.

Anyway, what do you all think? Would there be any interest for this to get implemented?

**Comment by listepo** [comment]
+1

**Comment by mikermcneil** [comment]
Thanks for posting, @Globegitter.  I'm an experimental issue cleanup bot-- nice to meet you!

It has been a couple of months since there have been any updates or new comments on this page.  If this issue has been resolved, please feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, etc.  Please include a link back to this page for reference.
  + add a comment to _this_ issue with a link to the new issue (for people arriving from search engines)

Thanks so much for your help!

<3
Sails.js proto-bot
